File: Recursion/recursion.java
package Recursion;

public class recursion {
    void fact(int fact){
        if(fact == 1){
            System.out.println(fact);
            return;
        }
        System.out.println(fact);
        fact(fact-1);
    }
    public static void main(String[] args) {
        recursion r = new recursion();
        r.fact(5); 
    }
}

File: Recursion/recursion.kt
class Recursion {
    fun fact(fact: Int) {
        if (fact == 1) {
            println(fact)
            return
        }
        println(fact)
        fact(fact - 1)
    }

    companion object {
        @JvmStatic
        fun main(args: Array<String>) {
            val r = Recursion()
            r.fact(5)
        }
    }
}
